plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -58,6 +58,12 @@ func init() {
 	// Cluster autoscaler friendly scheduling algorithm.
 	factory.RegisterAlgorithmProvider(ClusterAutoscalerProvider, defaultPredicates(),
 		replace(defaultPriorities(), "LeastRequestedPriority", "MostRequestedPriority"))
+
+	factory.RegisterPriorityMetadataProducerFactory(
+		func(args factory.PluginFactoryArgs) algorithm.MetadataProducer {
```

Comment:
oh yeah - I should have decribed it.
Basically, this is preparation to migrate SelectorSpreading to map-reduce-like framework:
https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go#L41

The problem is that for this priority, to avoid computing the same time multiple times, we need to precompute services of a given pod (for now also RCs and RSs, but this will change after migration to ownerref, but services still be there).
So we need some kind of ServiceLister to be available to the fuction computing metadata.

The most cleaner way I came up with (which is also consistent with priorities and predicates) is to take this lister from PluginFactoryArgs at the registration moment.

So as soon as this PR is merged (I wanted to avoid making it significantly larger), the returned MetadataProducer here will be PriorityMetadata which will be taking "serviceLister" (from PluginFactoryArgs) as an argument.

So it will be used in the next PR.